app/scripts/services/RegistrationCache.js

```diff
@@ -80,17 +80,19 @@ angular.module('confRegistrationWebApp')
       return defer.promise;
     };
 
-    this.getAllForConference = function (conferenceId, callback) {
+    this.getAllForConference = function (conferenceId, blocks) {
       var defer = $q.defer();
+      $rootScope.loadingMsg = 'Loading Registrations';
 
-      checkCache('conferences/' + conferenceId + '/registrations', function (registrations) {
-        angular.forEach(registrations, function (registration) {
-          update(path(registration.id), registration);
-        });
+      var blocksStr = '';
+      if(!_.isEmpty(blocks)){
+        blocksStr = '?block=' + blocks.join('&block=');
```

Comment:
I assume this function takes each element in the array and returns it as one string with `&block=` between each?